lib/ansible/modules/files/blockinfile.py

```diff
@@ -224,6 +224,9 @@ def main():
         if not module.boolean(params['create']):
             module.fail_json(rc=257,
                              msg='Path %s does not exist !' % path)
+        destpath = os.path.dirname(path)
+        if not os.path.exists(destpath) and not module.check_mode:
+            os.makedirs(destpath)
```

Comment:
This should probably have a `try`/`except` around it to handle the possible scenario of getting permission denied, and reporting that to the user appropriately instead of getting a stacktrace.

Reply:
Good observation. I have added a try/except to handle a possibile exception and pushed the changed in this PR. In this case the output will be something like the following:

$ ./bin/ansible localhost -m blockinfile -a 'dest=/tmp/ansible/.blockinfile/test state=present create=yes block=blah'
localhost | FAILED! => {
    "changed": false, 
    "failed": true, 
    "msg": "Error creating /tmp/ansible/.blockinfile Error code: 13 Error description: Permission denied"
}

I'm observing that also lineinfile module ( https://github.com/ansible/ansible/blob/cee5d420de79c51caf450801b7a3d0bb8f1cde02/lib/ansible/modules/files/lineinfile.py#L241 ) maybe suffers the same problem.